### PR TITLE
Implement LWG-3712: `chunk_view` and `slide_view` should not be `default_initializable`

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Towards #2872.

### CODE REVIEW

File: tests/std/tests/P2442R1_views_chunk/test.cpp [hunk 1]
```diff
@@ -35,6 +35,9 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
     STATIC_ASSERT(forward_range<R> == forward_range<Rng>);
     STATIC_ASSERT(bidirectional_range<R> == bidirectional_range<Rng>);
     STATIC_ASSERT(random_access_range<R> == random_access_range<Rng>);
+    // Validate non-default-initializability
+    STATIC_ASSERT(!is_default_constructible_v<R>);
+    STATIC_ASSERT(!default_initializable<R>);
```
Comment (StephanTLavavej) [inline tests/std/tests/P2442R1_views_chunk/test.cpp:40]: No change requested: `test_one()` is called for both input-only and forward ranges, so this covers the LWG issue's modifications to both the primary template and the specialization for forward ranges.

**Review by StephanTLavavej — APPROVED** [review]

File: tests/std/tests/P2442R1_views_chunk/test.cpp (hunk 1 above)
Comment (CaseyCarter) [inline tests/std/tests/P2442R1_views_chunk/test.cpp:40]: A type that cannot be value-initialized certainly cannot be default-initialized, so `!is_default_constructible_v<T>` implies `!default_initializable<T>` - we don't need to redundantly check both. I'll cleanup.

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing so many LWG issue resolutions and improving conformance! :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat:

**Pull request merged**